src/custom-apps/components/PermissionAlert/utils.ts

```diff
@@ -1,47 +1,53 @@
-// @ts-strict-ignore
-import { gql } from "@apollo/client";
-import { FieldNode, parse, SelectionNode, visit } from "graphql";
-
-interface IntrospectionNode {
-  kind: string;
-  name: string;
-  description: string;
-  fields: Array<{
-    name: string;
-    description: string;
-  }>;
-}
+import {
+  buildClientSchema,
+  DocumentNode,
+  GraphQLField,
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLSchema,
+  GraphQLType,
+  IntrospectionQuery,
+  Location,
+  parse,
+  Token,
+  visit,
+} from "graphql";
+import { TypeMap } from "graphql/type/schema";
+
+type SubscriptionQueryPermission = Record<
+  string,
+  {
+    isOneOfRequired: boolean;
+    permissions: string[];
+  }
+>;
 
-interface PermissionChildNode {
-  permissions: string[];
-  children: string[];
-}
+const getStartToken = (documentNode: DocumentNode) => {
+  const loc = documentNode.loc as Location | undefined;
 
-interface PermissionNode {
-  permissions: string[];
-  children: Record<string, PermissionChildNode>;
-}
+  return loc?.startToken;
+};
 
-type PermissionMap = Record<string, PermissionNode>;
-
-// cannot be in `queries.ts` as codegen cannot handle `__schema`
-export const IntrospectionQuery = gql`
-  query PermissionIntrospection {
-    __schema {
-      types {
-        kind
-        name
-        description
-        fields(includeDeprecated: false) {
-          name
-          description
-        }
-      }
+const tokenTree = (token: Token | undefined, tokens: string[] = []): string[] => {
```

Comment:
question: What does this function do? IMO it is not a token tree it is more like flattened token tree